src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,48 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private readonly Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(_headers is null)
+                {
+                    return Array.Empty<IHttpCookie>();
+                }
```

Comment:
Wouldn't this cause it to always return an empty array because the `_headers` property will be null until the `Headers` (public) property is accessed ? Did you mean to use `Headers` here instead of `_headers` ?